### ISSUE TITLE
Add ability to change blend equations (glBlendEquation and glBlendEquationSeparate in OpenGL)

### ISSUE THREAD

**Issue by dt-universe** [YoYoGames/GameMaker-Feature-Requests#1404]
### Is your feature request related to a problem?

When I try to generate mask surface from several mask sprites with smooth edges, I have visual artifacts when they overlap. By default GameMaker uses ADD operation between Source and Destination colors, in my case I need MAX operation between them.

![blend_equation](https://github.com/YoYoGames/GameMaker-Feature-Requests/assets/23583854/2d6e18e8-bbf2-441e-9fb5-789b08de77af)

### Describe the solution you'd like

Add new gpu_blendmode_equation and gpu_blendmode_equation_sepalpha functions, which will change operation between Source and Destination colors. In OpenGL these functions are glBlendEquation and glBlendEquationSeparate.

### Describe alternatives you've considered

I know just one solution to this issue, with 3 (!) screen-space surfaces, one for drawing current sprite on it, and two for ping-pong, in final one of them will be result, which we will draw on screen. If you want to see whole pipeline for this effect, you can check sample project below, vk_space changing from default ADD equation to "emulation" of MAX equation. "Emulation" is so expensive (different surfaces, different blend modes, shader). Will be so cool to have native hardware related functions.

[blend_equation.zip](https://github.com/YoYoGames/GameMaker-Feature-Requests/files/13061924/blend_equation.zip)

**Comment by stuckie** [comment]
You should already be able to do this via the gpu_set_blendmode* functions.

**Comment by dt-universe** [comment]
No, it's not possible to make this with blend factors, as I know, if you know how, may you give me an example? If you are talking about bm_max blend mode, then no, it's not the same and just misnaming. For example, if you have Source color (1, 0, 0, 0.5) and Destination color (0, 0, 0, 0.1), then:

```
gpu_set_blendmode(bm_max) and ADD (default) equation
(0.5, 0, 0, 0.25) + (0, 0, 0, 0.05) = (0.5, 0, 0, 0.3)

gpu_set_blendmode_ext(bm_one, bm_one) and MAX equation
MAX((1, 0, 0, 0.5), (0, 0, 0, 0.1)) = (1, 0, 0, 0.5)
```

**Comment by dt-universe** [comment]
> You should already be able to do this via the gpu_set_blendmode* functions.

No, it's not possible to make this with blend factors, as I know, if you know how, may you give me an example? If you are talking about bm_max blend mode, then no, it's not the same and just misnaming. For example, if you have Source color (1, 0, 0, 0.5) and Destination color (0, 0, 0, 0.1), then:

```
gpu_set_blendmode(bm_max) and ADD (default) equation
(0.5, 0, 0, 0.25) + (0, 0, 0, 0.05) = (0.5, 0, 0, 0.3)

gpu_set_blendmode_ext(bm_one, bm_one) and MAX equation
MAX((1, 0, 0, 0.5), (0, 0, 0, 0.1)) = (1, 0, 0, 0.5)
```

All modern engines and frameworks support this feature.

**Comment by rwkay** [comment]
OK looking at the documentation here https://registry.khronos.org/OpenGL-Refpages/es2.0/ GL_MAX is not available for Open GL ES 2.0 and I would need to investigate how well it is supported across the various platforms, this will not be done on the current runtime - but checking webgpu and the blend operation can definitely be set on a pipeline there https://www.w3.org/TR/webgpu/#enumdef-gpublendoperation so we will be supporting it on the new runtime.

**Comment by rwkay** [comment]
This feature will be in the new runtime we will not be adding it to the current runtime.
